Update web app manifest page and add missing `file_handlers` member (#25222)
mdn/content: files/en-us/web/manifest/index.md

{{QuickLinksWithSubpages("/en-US/docs/Web/Manifest")}}

**Web app manifests** are part of a collection of web technologies called [progressive web apps](/en-US/docs/Web/Progressive_web_apps) (PWAs).

> PWAs are web apps that can be installed on a device without an app store. Unlike browser or home screen bookmarks, installed PWAs can work offline, use regular [Web APIs](/en-US/docs/Web/API), and integrate in the operating system they're installed on more deeply than websites can.

Reviewer: ```suggestion
PWAs are web apps that can be installed on a device without an app store. Unlike browser or home screen bookmarks, installed PWAs can work offline, can use regular [Web APIs](/en-US/docs/Web/API) while offline, and can be fully integrated into the operating system they're installed on.
```

> PWAs are web applications, written using web technologies, that can be installed on a device without an app store. Unlike browser or home screen bookmarks, installed PWAs can work offline, can use regular [Web APIs](/en-US/docs/Web/API), and can be fully integrated into the operating system they're installed on.

Author: Fully agreed. Making the change now.

A web application manifest, as defined in the [Web Application Manifest](https://w3c.github.io/manifest/) specification, provides information about a web application in a {{Glossary("JSON")}} text file. A web application manifest is necessary for the web app to be installed on a device and behave like other OS-native apps.

> A PWA's manifest includes its name, icon(s), description, and ways that the PWA appears and integrates in the operating system where it's installed.

Reviewer: ```suggestion
A PWA's manifest includes its [name]((/en-US/docs/Web/Manifest/name), [icon(s)](/en-US/docs/Web/Manifest/icons), [description](/en-US/docs/Web/Manifest/description), and ways that the PWA [appears](/en-us/web/progressive_web_apps/how_to/customize_your_app_colors) and integrates into the operating system where it's installed.
```

Reviewer: we should link to the how tos once created.

Author: Although our How-to pages do not have any sidebars yet. So if we link to them from visible content today, that means people will end up on page that they can't "escape" from easily (no consistent sidebar to come back to the general manifest page). Won't that be a problem?

Reviewer 2: Yes, I agree that's a problem. Can we defer cross-links like this until we have the pieces and nav all in place? We can do a last pass through everything and fix up things like this then.

## Members

> A web application manifest contains a single JSON object where the top-level keys are called _members_. The members that a web application manifest can contain are listed below. Click on each one to find more information about it:

Reviewer: thank you! This definition was so needed.

> A web application manifest contains a single JSON object where the top-level keys are called _members_. The members that a web application manifest can contain are listed below. Click on each one to find more information about it:

Reviewer: ```suggestion
A web application manifest contains a single JSON object where the top-level keys are called _members_. The following are members that a web application manifest can contain:
```

## Splash screens

> In some browsers and operating systems, a splash screen is displayed when an installed PWA is launched. This splash screen is auto-generated from members in the web app manifest, specifically:

Reviewer: ```suggestion
In some browsers and operating systems, a splash screen is displayed when an installed PWA is launched. This splash screen is automatically generated with its appearance defined by members in the web app manifest, specifically:
```

Author: I don't think there was ever a spashscreen on desktop operating systems, there also doesn't appear to be one on iOS devices. The only place that I think ever had a splashscreen was Android. But testing just now on an emulator, I don't it exists anymore. It might have been there in previous versions, but has apparently been removed. I'll try to ask around and see if I can find out what happened there.

Reviewer: found references https://stackoverflow.com/questions/3877049/iphone-web-app-splash-screen-delay, - the "apple-touch-startup-image" - good to know i am not off my rocker

Author: Thanks for finding this. If I understand correctly, iOS devices do generate splash screens using `link` tags in your HTML.

I have also asked on the Chromium Slack. While I haven't heard back anything super conclusive yet, it does seem that splash screens also exist on Android devices.

I will apply your suggested change in this PR, but my goal isn't to document how to make a PWA splash screen here. This would actually make a good new How-to page.
